pkg/platform/runtime/buildexpression/buildexpression.go

```diff
@@ -171,6 +171,37 @@ func New(data []byte) (*BuildExpression, error) {
 	return expr, nil
 }
 
+// NewFromPlatformIDAndRequirements creates a minimal buildexpression from the given platform UUID
+// and requirement list.
+func NewFromPlatformIDAndRequirements(platformID string, requirements []model.Requirement) (*BuildExpression, error) {
```

Comment:
This is really specific to the current use case. Do you think we could make this more generic? It could return a blank build expression and then we leave it up to the caller to update any requirements and timestamps.